autograder/plagiarism_detection/plagiarism_detection.py

```diff
@@ -1,50 +1,96 @@
-import sys
+import sys, math
 from antlr4 import *
 from lexers.Java8Lexer import JavaLexer
 from lexers.Python3Lexer import Python3Lexer
 
+
 def compare(files):
     numFiles = len(files)
     # add error handling
     if numFiles == 0:
         return "No files found"
-    prev = files[0].name.split('.')[1]
+    lexerClass, ignoreList, numTokens = checkFileTypes(files)
+    similarityScores = [[0 for x in range(numFiles)] for y in range(numFiles)]
+    # get token stream for each file and total frequency for each token type
+    tokenStreams, tokenFrequencies = parseFiles(
+        files, lexerClass, ignoreList, numTokens
+    )
+    similarityMatrix = buildSimilarityMatrix(tokenFrequencies)
+    print(similarityMatrix)
     for i in range(numFiles):
-        fileType = files[i].name.split('.')[1]
+        for j in range(i + 1, numFiles):
+            similarity = getSimilarity(tokenStreams[i], tokenStreams[j])
+            similarityScores[i][j] = similarity
+            similarityScores[j][i] = similarity
+    return similarityScores
+
+
+# lexerClass is the lexer to be used, ignoreList is a list containing the
+# token types to remove when parsing, numTokens is the number of token types in the language
+def checkFileTypes(files):
```

Comment:
Please, use snake_case for both function and variable names as described in PEP8.